modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1916,13 +1918,114 @@ def sort_by(
+
+        def sort_function(df):
+            index_renaming = None
+            if any(name in df.columns for name in df.index.names):
+                index_renaming = df.index.names
+                df.index = df.index.set_names([None] * len(df.index.names))
+            df = df.sort_values(by=columns, ascending=ascending, **kwargs)
+            if index_renaming is not None:
+                df.index = df.index.set_names(index_renaming)
+            return df
+
+        axis = Axis(axis)
+        # This if selects cases where we are either sorting by a single column that is not
+        # the index, or we are sorting by multiple columns, since in this case, we will have to
+        # shuffle the data before sorting.
+        if axis == Axis.ROW_WISE and not (
+            len(columns) == 1 and columns[0] == "__index__"
+        ):
+            # If this df is empty, we don't want to try and shuffle or sort.
+            if len(self.axes[0]) == 0 or len(self.axes[1]) == 0:
+                return self.copy()
+            # If this df only has one row partition, we don't want to do a shuffle and sort - we can
+            # just do a full-axis sort.
+            if len(self._partitions) == 1:
+                return self.apply_full_axis(
+                    1,
+                    sort_function,
+                )
+            if self.dtypes[columns[0]] == object:
+                # This means we are not sorting numbers, so we need our quantiles to not try
+                # arithmetic on the values.
+                method = "inverted_cdf"
+            else:
+                method = "linear"
+            from .utils import build_sort_functions
+
+            shuffling_functions = build_sort_functions(
+                self, columns, method, ascending, **kwargs
+            )
+            new_partitions = self._partition_mgr_cls.shuffle_partitions(
+                self._partitions,
+                shuffling_functions["sample_function"],
+                shuffling_functions["pivot_function"],
+                shuffling_functions["split_function"],
+                sort_function,
+            )
+            new_axes = self.axes
+            if kwargs.get("ignore_index", False):
+                new_axes[axis.value] = RangeIndex(len(new_axes[axis.value]))
+            else:
+                new_axes[axis.value] = self._compute_axis_labels_and_lengths(
+                    axis.value, new_partitions
+                )[0]
+            if isinstance(self.axes[axis.value], pandas.MultiIndex):
+                new_axes[axis.value] = pandas.MultiIndex.from_tuples(
+                    new_axes[axis.value].values
+                )
+            new_axes[axis.value] = new_axes[axis.value].set_names(
+                self.axes[axis.value].names
+            )
+            new_axes[axis.value ^ 1] = self.axes[axis.value ^ 1]
+            new_lengths = [None, None]
+            (
+                new_partitions,
+                new_lengths[axis.value],
```

Comment:
Is there no way to compute `new_lengths[axis.value ^ 1]`

Reply:
Technically we know what the new_lengths are - since we're axis partitioned, its just going to be a list of one - the width of the df. Would you like me to manually specify this for `new_lengths`?

Reply:
Changed so we manually compute `new_lengths[axis.value ^ 1]`.